specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-06-01-preview/SourceControls.json

```diff
@@ -404,8 +417,41 @@
       ],
       "type": "object"
     },
+    "RepositoryAccess": {
+      "type": "object",
+      "properties": {
+        "kind": {
+          "description": "The kind of repository access credentials",
+          "$ref": "#/definitions/RepositoryAccessKind"
+        },
+        "code": {
+          "type": "string",
+          "description": "OAuth Code. Required when `kind` is `OAuth`"
+        },
+        "state": {
+          "type": "string",
```

Comment:
>string [](http://example.com/codeflow?start=19&length=6)

Should this be an enum?

Reply:
This is OAuth state, which may be an arbitrary value. From GitHub definition: 

> An unguessable random string. It is used to protect against cross-site request forgery attacks.